test/parachain/tests/integration_test.rs

```diff
@@ -0,0 +1,411 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+// TODO: this is necessary for the jsonrpsee macro used
+#![allow(unused_variables, dead_code)]
+
+use assert_cmd::cargo::cargo_bin;
+use codec::Encode;
+use polkadot_primitives::parachain::{Info, Scheduling};
+use polkadot_primitives::Hash as PHash;
+use polkadot_runtime::{Header, OnlyStakingAndClaims, Runtime, SignedExtra, SignedPayload};
+use polkadot_runtime_common::{parachains, registrar, BlockHashCount};
+use regex::Regex;
+use sp_arithmetic::traits::SaturatedConversion;
+use sp_runtime::generic;
+use sp_version::RuntimeVersion;
+use std::io::Read;
+use std::{
+	convert::TryInto,
+	env, fs, io, net,
+	path::PathBuf,
+	process::{Child, Command, Stdio},
+	thread,
+	time::Duration,
+};
+use substrate_test_runtime_client::AccountKeyring::Alice;
+use tempfile::tempdir;
+
+static POLKADOT_ARGS: &[&str] = &["polkadot", "--chain=res/polkadot_chainspec.json"];
+
+jsonrpsee::rpc_api! {
+	Author {
+		#[rpc(method = "author_submitExtrinsic", positional_params)]
+		fn submit_extrinsic(extrinsic: String) -> PHash;
+	}
+
+	Chain {
+		#[rpc(method = "chain_getFinalizedHead")]
+		fn current_block_hash() -> PHash;
+
+		#[rpc(method = "chain_getHeader", positional_params)]
+		fn header(hash: PHash) -> Option<Header>;
+
+		#[rpc(method = "chain_getBlockHash", positional_params)]
+		fn block_hash(hash: Option<u64>) -> Option<PHash>;
+	}
+
+	State {
+		#[rpc(method = "state_getRuntimeVersion")]
+		fn runtime_version() -> RuntimeVersion;
+	}
+}
+
+// Adapted from
+// https://github.com/rust-lang/cargo/blob/485670b3983b52289a2f353d589c57fae2f60f82/tests/testsuite/support/mod.rs#L507
+fn target_dir() -> PathBuf {
+	env::current_exe()
+		.ok()
+		.map(|mut path| {
+			path.pop();
+			if path.ends_with("deps") {
+				path.pop();
+			}
+			path
+		})
+		.unwrap()
+}
+
+struct ChildHelper<'a> {
+	name: String,
+	child: &'a mut Child,
```

Comment:
Yes it started differently, not really as a helper... I will check that out

Reply:
I think I will keep that one as it as I think this is not meant to wrap and hide the child process. It might be useful for the tester to still be able to access the child. I did rename the variable to make it clear it's not the child but the helper